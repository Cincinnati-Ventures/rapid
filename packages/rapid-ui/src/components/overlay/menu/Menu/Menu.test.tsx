import React from 'react';
import { render } from '@testing-library/react';
import { expect, it, describe } from 'vitest';
import { Menu, MenuItem, MenuItems, MenuButton } from '..';

describe('Menu', () => {
	it('should render default correctly', () => {
		const { container } = render(
			<Menu>
				<MenuItems>
					<MenuItem />
				</MenuItems>
				<MenuButton />
			</Menu>,
		);
		expect(container).toMatchSnapshot();
	});

	it('should render correctly with non default props', () => {
		const { container } = render(
			<Menu styles='bg-blue-500'>
				<MenuItems styles='bg-blue-500'>
					<MenuItem styles='bg-blue-500'>Menu Item</MenuItem>
					<MenuItem styles='bg-blue-500'>Menu Item</MenuItem>
				</MenuItems>
				<MenuButton styles='bg-blue-500' />
			</Menu>,
		);
		expect(container).toMatchSnapshot();
	});
});
