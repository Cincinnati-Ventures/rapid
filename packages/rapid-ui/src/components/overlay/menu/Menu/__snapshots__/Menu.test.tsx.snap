// Vitest Snapshot v1

exports[`Menu > should render correctly with non default props 1`] = `
<div>
  <div
    class="relative inline-block text-left bg-blue-500 rapid-menu"
    data-headlessui-state=""
  >
    <button
      aria-expanded="false"
      aria-haspopup="menu"
      class="rapid-menu-button bg-blue-500"
      data-headlessui-state=""
      id="headlessui-menu-button-:r1:"
      type="button"
    />
  </div>
</div>
`;

exports[`Menu > should render default correctly 1`] = `
<div>
  <div
    class="relative inline-block text-left rapid-menu"
    data-headlessui-state=""
  >
    <button
      aria-expanded="false"
      aria-haspopup="menu"
      class="rapid-menu-button"
      data-headlessui-state=""
      id="headlessui-menu-button-:r0:"
      type="button"
    />
  </div>
</div>
`;
